sdk/state/open.go

```diff
@@ -90,14 +103,40 @@ func (c *Channel) openTxs(d OpenAgreementDetails) (decl, close, formation *txnbu
 // be used prior to prepare an open agreement with the other participant.
 func (c *Channel) OpenTx() (formationTx *txnbuild.Transaction, err error) {
 	openAgreement := c.openAgreement
-	_, _, formationTx, err = c.openTxs(openAgreement.Details)
+	declTx, closeTx, formationTx, err := c.openTxs(openAgreement.Details)
 	if err != nil {
-		return nil, fmt.Errorf("building declaration and close txs for latest close agreement: %w", err)
+		return nil, fmt.Errorf("building txs for for open agreement: %w", err)
 	}
 	formationTx, err = formationTx.AddSignatureDecorated(openAgreement.FormationSignatures...)
 	if err != nil {
 		return nil, fmt.Errorf("attaching signatures to formation tx for latest close agreement: %w", err)
 	}
+	for _, s := range openAgreement.DeclarationSignatures {
```

Comment:
💡 for these steps of finding the specific signature I think we should add comments. w/o context it's not completely clear why we have to do this imo

Reply:
Good point. This code was gnarly enough to author let alone read. 07d1f8c